### PR TITLE
Adding Me Protocol To Grants-Program

### PR DESCRIPTION
**Pull request by AgentPoles**
#### Project Abstract

> The Me Protocol is a purpose-built exchange infrastructure for blockchain-based loyalty rewards. It provides the fundamental technology for the creation of a frictionless, limitless, and decentralised rewards ecosystem for all companies, large or small, to participate.

#### Grant [level](https://github.com/w3f/Grants-Program#level_slider-levels)

- [ ] **Level 1**:  Up to $10,000, 2 approvals
- [x] **Level 2**:  Up to $30,000, 3 approvals
- [ ] **Level 3**:  Unlimited, 5 approvals (for >$100k: Web3 Foundation Council approval)

#### Application Checklist

- [x] The [application template](https://github.com/w3f/Grants-Program/blob/master/applications/application-template.md) has been copied and aptly renamed (`MeProtocol.md`).
- [x] I have read the [application guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/grant_guidelines_per_category.md).
- [x] A BTC, Ethereum (USDC/DAI) or Polkadot/Kusama (aUSD/USDT) address for the payment of the milestones is provided in the application.
- [x] The software delivered for this grant will be released under an open-source license specified in the application.
- [x] The initial PR contains only one commit (squash and force-push if needed).
- [x] The grant will only be announced once the first milestone [has been accepted](https://github.com/w3f/Grant-Milestone-Delivery#process) (see the [announcement guidelines](https://github.com/w3f/Grants-Program/blob/master/docs/Support%20Docs/announcement-guidelines.md)).
- [x] I prefer the discussion of this application to take place in a private Telegram channel. My username is: `@meprotocol`

### CODE REVIEW

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the new application, but it’s usually easier for us to keep track of the changes if you continue to work on the previous application (https://github.com/w3f/Grants-Program/pull/1332) and not immediately close it. That said, the application looks good now, it just would be nice if you could make the specification of the milestone tables a little bit more concrete. For example, add the programming/smart contract language to each delivery and define the functions a little bit more concretely. Also, something like “we will iteratively test the entire solution to achieve suitability” is not something that we can easily evaluate or test. So feel free to remove it.

**Comment by AgentPoles** [comment]
Hello @Noc2, Thanks a lot for your feedback. 
I have updated the file as requested and I am open to making more changes to ensure the milestones are clear enough.
Thank you very much!

**Review by Noc2 — DISMISSED** [review]
Thanks for the update. Looks good to me. I'm happy to go ahead with it and share it with the rest of the team.

File: applications/MeProtocol.md
Comment (alxs) [inline applications/MeProtocol.md:6]:
```suggestion
- **Level:** 2 
```

**Comment by alxs** [comment]
Thank you for the application @AgentPoles. Please note that deliverables need to be concrete, verifiable and potentially useful to others in the ecosystem. I.e. we can't fund things like "[Setting up](https://github.com/w3f/Grants-Program/pull/1340/files#diff-434d715a966556e2550380fe0b388704bdbb5cc8916a033aa1087b5277ad7c01R202)", "[Testing](https://github.com/w3f/Grants-Program/pull/1340/files#diff-434d715a966556e2550380fe0b388704bdbb5cc8916a033aa1087b5277ad7c01R208)" or "Connect APP to Protocol".  Please use the deliverable table to list and describe the components you will be developing and their functionality. See also our [FAQ](https://w3f.github.io/Grants-Program/FAQ#what-activitiespositions-do-you-fund) in this regard.

**Comment by Noc2** [comment]
@AgentPoles I’m closing this due to inactivity. Let me know if I should re-open it.

**Comment by AgentPoles** [comment]
Hello @Noc2 and other Grant committee members, please so sorry for the delay in response, we have been working on the updates locally and will be providing the changes as requested. We will be glad to have it re-opened. Thanks a lot!

**Comment by Noc2** [comment]
@AgentPoles, that's great to hear. I reopened it.

**Comment by AgentPoles** [comment]
@Noc2  and @alxs thanks a lot!
We have updated the changes as requested. We are open to more corrections till we satisfy all review conditions. Thanks a lot for your patience and sorry for the back and forth thus far.

**Review by Noc2 — CHANGES_REQUESTED** [review]
Thanks for the update. Just two small additional comments: 
- Could remove the mention of any specific token from your application, see our guidelines: https://github.com/w3f/Grants-Program#guidelines 
- Could you remove the three dots and the last row of your last milestone table?

**Comment by AgentPoles** [comment]
Hello @Noc2,  Thanks a lot for your feedback. We have made the requested changes and are open to more changes. Also, thanks a lot for your help, we really appreciate it.

**Review by Noc2 — APPROVED** [review]
Thanks for the update. I mark it again as ready for review and I'm still happy to go ahead with it.

**Review by gautamdhameja — APPROVED** [review]

**Review by randombishop — APPROVED** [review]
I like this application, approved from my end.

**Review by keeganquigley — APPROVED** [review]
Thanks for the interesting application @AgentPoles I think this is a unique idea, though I have some initial doubts over how well this would go over with the brand's customer base, since the customer has to have a wallet to use the app. But overall I think these smart contracts will be useful to the ecosystem and since you want to eventually graduate to pallets/parachains I am happy to support this initial concept.

**Comment by AgentPoles** [comment]
Many thanks for everyone’s support, we are excited to get started on the grant and support the growth of the ecosystem!
Thank you very much!

**Comment by keeganquigley** [comment]
Hi @AgentPoles are you able to give us an update on M1?

**Comment by AgentPoles** [comment]
Hi @keeganquigley yes please! we will be dropping updates any time next week or the upper one. We have some implementations but currently having some setbacks with OpenBrush (the ink! framework we are using). We have been checking in with their team on discord and we will get the updates rolled out ASAP. Thanks a lot!

**Comment by keeganquigley** [comment]
Hi @AgentPoles are you planning on delivering a milestone anytime soon? If not we would ask that an [amendment](https://github.com/w3f/Grants-Program#changes-to-a-grant-after-approval) be submitted to extend the timeline. Let us know, thanks!

**Comment by AgentPoles** [comment]
Hi @keeganquigley yes, we have been working on this heavily... still fighting through all the drawbacks we have with the tooling, code implementations are done, we are in the testing phase but our hire is currently ill, we are looking to add someone new to our team to speed things up, would be glad to get help on this if you have any. I can  give you access to our repo so you track our daily/weekly progress. Also happy to discuss the  amendments cause we sincerely want to see this through till the end. Thanks a lot!

**Comment by keeganquigley** [comment]
Thanks @AgentPoles for the update! No need to add me to the repo, as we don't track daily/weekly progress but just want to make sure it is still being worked on. Delays aren't an issue, but if you aren't planning on submitting in the next few weeks I'd suggest filing a PR to update the estimated duration dates on your application.

As far as help goes, you could try reaching out to [ShellBoxes](https://www.shellboxes.com/#blockchain), a development firm, as they contacted me to let me know that they provide technical assistance for projects. However be aware that their service does cost $ after the initial consultation. Thanks!

**Comment by AgentPoles** [comment]
Alright @keeganquigley thanks a lot, we will discuss internally, take the next appropriate steps and update you. Thank you very much!

**Comment by keeganquigley** [comment]
pinging @AgentPoles

**Comment by AgentPoles** [comment]
Hi @keeganquigley we are in the rehiring phase currently and have prospective candidates, Non have been impressive yet. Also we are also considering making adjustments to the milestones so we can deliver the parts we have successfully implemented and hence track some progress on the milestones. Please what is your take on this?

**Comment by keeganquigley** [comment]
Thanks @AgentPoles sorry for the delay; I was out of office for a couple weeks. Let me discuss this with the team internally and get back to you.

**Comment by keeganquigley** [comment]
Hi @AgentPoles sorry for the delay, I did talk with the team about this. Feel free to amend the original application and submit an amendment PR to adjust the milestones to deliver what you have so far. You can also document any staffing changes. The amendment will have to be approved by the committee, but given the changes are out of your control, I would believe this should happen relatively quickly. Thanks!

**Comment by AgentPoles** [comment]
HI @keeganquigley thank you so much, really grateful for you patience and  persistent help. 

We have successfully secured a new Hire, so I guess we will continue with the original plan, although some things have changed, but the core items for delivery are still preserved. We hope to get the first milestone rolled out in this month (as agreed with the hire). We will be adjusting the delivery dates to reflect this updates. Thank you so so much!

**Comment by keeganquigley** [comment]
Thanks @AgentPoles are you planning to submit the amendment to reflect the new delivery dates?

**Comment by keeganquigley** [comment]
Hi @AgentPoles seeing as it has been almost a year, I think it is probably best to go ahead and close this grant for now, and you can always potentially pick it up again in the future once your staffing issues have been figured out. Let me know if that sounds good. Otherwise we will go ahead and close it in two weeks. Thank you!

**Comment by AgentPoles** [comment]
HI @keeganquigley, sorry i missed your previous comment, I was out of office. We have resolved the hire issue as i mentioned earlier and we have now successfully completed the first milestone. I Just got back this week and I am currently reviewing and preparing the delivery.

**Comment by keeganquigley** [comment]
Glad to hear it @AgentPoles thanks for the update!

**Comment by AgentPoles** [comment]
Hi @keeganquigley we are set to deliver now, but please we have a final question on licensing. We will like to keep our project open-sourced, but restrict commercial use for a time frame, please what are your views on this and is it allowed?

**Comment by AgentPoles** [comment]
Hi @keeganquigley just checking in on this, thank a lot!

**Comment by AgentPoles** [comment]
yes @keeganquigley. Please let us know, and thank you very much for your support all through!

**Comment by keeganquigley** [comment]
Np @AgentPoles I talked to the team and unfortunately we are unable to accept this type of license. Basically, our program is pretty strict about being open-source. Let me know how you would like to proceed.

**Comment by AgentPoles** [comment]
Alright @keeganquigley,  thank you very much. Let me discuss with my team and revert back to you also.

**Comment by AgentPoles** [comment]
Hi @keeganquigley sorry for the delays we are finalizing on this with all necessary team members and stakeholders

**Comment by AgentPoles** [comment]
Hi @keeganquigley, we have decided to proceed with the GPL v3 License. We are set to raise our PR. Also please, we will like to update the wallet address provided in this application. Do we do that first before raising the delivery PR (as i can see we need to specify the application PR Number when submitting the delivery)? or we can do both simultenously ?

**Comment by keeganquigley** [comment]
Thanks @AgentPoles for the update. To update the wallet address, please amend the original application and submit a PR, as it will require 3 approvals. But yes you can submit the delivery in the meantime, it is not a problem to submit both simultaneously. Thanks!

**Comment by AgentPoles** [comment]
Alright @keeganquigley, thanks a lot! Made the PR for the delivery, will now proceed to update the application.

**Comment by keeganquigley** [comment]
Thanks @AgentPoles yes if you could that will allow us to update our records, and that way you can extend the other milestones as well. Thanks!

**Comment by AgentPoles** [comment]
@keeganquigley raised a PR to update the grant application. Thank you very much!

**Comment by AgentPoles** [comment]
Hello @keeganquigley, Just checking in for any updates on the milestone 1 delivery. Thanks a lot!

**Comment by AgentPoles** [comment]
Hello @keeganquigley, Just checking in for any updates on the milestone 1 delivery. Thanks a lot!

**Comment by keeganquigley** [comment]
@AgentPoles hi are you able to give an update on milestone 2?

**Comment by AgentPoles** [comment]
Hi @keeganquigley,  thanks a lot for reaching out. Yes please,  we are closing out on milestone 2 very soon, working on the test cases now. Thank you very much!

**Comment by AgentPoles** [comment]
Hi @keeganquigley, please we just raised a PR for Milestone 2 Delivery, thank you very much!

**Comment by keeganquigley** [comment]
Hi @AgentPoles just checking in on M3. Thanks!

**Comment by keeganquigley** [comment]
pinging @AgentPoles are you still working on milestone 3? Thanks

**Comment by AgentPoles** [comment]
Hi @keeganquigley yes please we are still working on M3, we are making progress and will be closing it out soon!

**Comment by keeganquigley** [comment]
Hi @AgentPoles are you able to provide a status update?

**Comment by keeganquigley** [comment]
pinging @AgentPoles

**Comment by keeganquigley** [comment]
Hi @AgentPoles please let us know if you are still working on this grant, otherwise we will close it in the coming days due to inactivity. Thanks.

**Pull request merged**
